src/components/Palette/Palette.tsx

```diff
@@ -0,0 +1,287 @@
+
+    const onFocusGrid = React.useCallback(() => {
+        focusOnOptionWithIndex(0);
+    }, [focusOnOptionWithIndex]);
+
+    const onBlurGrid = React.useCallback(() => {
+        setFocusedOptionIndex(undefined);
+    }, []);
+
+    const onFocusOption = React.useCallback(
+        (event: React.FocusEvent<HTMLButtonElement>) => {
+            const newIndex = options.findIndex((option) => option.value === event.target.value);
+            if (newIndex >= 0) {
+                focusOnOptionWithIndex(newIndex);
+            }
+
+            onFocus?.(event);
+            event.stopPropagation();
+        },
+        [options, focusOnOptionWithIndex, onFocus],
+    );
+
+    const onBlurOption = React.useCallback(
+        (event: React.FocusEvent<HTMLButtonElement>) => {
+            onBlur?.(event);
+            setFocusedOptionIndex(undefined);
+            event.stopPropagation();
+        },
+        [onBlur],
+    );
+
+    return (
+        <div
+            {...containerProps}
+            role={'grid'}
+            ref={handleRef}
+            className={paletteClassNames.palette({size}, className)}
+            onFocus={focusedOptionIndex === undefined ? onFocusGrid : undefined}
+            onBlur={onBlurGrid}
+            style={style}
+            data-qa={qa}
+            tabIndex={focusedOptionIndex === undefined ? 0 : -1}
+        >
+            {rows.map((row, rowNumber) => (
+                <div
+                    className={paletteClassNames.row(rowClassName)}
+                    key={`row-${rowNumber}`}
+                    role={'row'}
+                >
+                    {row.map((option) => (
+                        <PaletteOptionItem
+                            key={option.value}
+                            value={currentValue}
+                            option={option}
+                            checked={currentValue?.includes(option.value) ?? false}
+                            disabled={disabled}
+                            size={size}
+                            optionClassName={optionClassName}
+                            iconClassName={iconClassName}
+                            onUpdate={setCurrentValue}
+                            onFocus={onFocusOption}
+                            onBlur={onBlurOption}
+                        />
+                    ))}
+                </div>
+            ))}
+        </div>
+    );
+}) as PaletteComponent;
+
+Palette.displayName = 'Palette';
+
+function getRows(options: PaletteOption[], columns: number): PaletteOption[][] {
+    const rows: PaletteOption[][] = [];
+    let row: PaletteOption[] = [];
+
+    let column = 0;
+    for (const option of options) {
+        row.push(option);
+        column += 1;
+        if (column === columns) {
+            rows.push(row);
+            row = [];
+            column = 0;
+        }
+    }
+
+    if (row.length > 0) {
+        rows.push(row);
+    }
+
+    return rows;
+}
+
+function PaletteOptionItem({
```

Comment:
Suggestions about `PaletteOptionItem` component:
- rename ~~PaletteOptionItem~~ -> `PaletteOption`
- use `Button` as wrapper for option (you're already doing it) and add property `renderOption` (like in `Select`) and use `Button.Icon` by default. In this case, we will not need the following options: `optionClassName`, `iconClassName`
- rename `checked` -> selected